src/main/java/org/fcrepo/spec/testsuite/crud/ExternalBinaryContent.java

```diff
@@ -464,6 +927,87 @@ public void binaryContentRedirectStatusHead(final String uri) {
+
+        doHeadUnverified(location, true).then()
+                .statusCode(anyOf(is(302), is(307)));
+    }
+
+    private boolean externalContentSupported(final String uri) {
+        if (extContentSupported == null) {
+            final String header = doOptions(uri).header("Accept-External-Content-Handling");
+            extContentSupported = header != null;
+        }
+
+        return extContentSupported;
+    }
+
+    private List<String> getAcceptedHandlingTypes(final String uri) {
+        final String header = doOptions(uri).header("Accept-External-Content-Handling");
+        if (header == null) {
+            return emptyList();
+        }
+        return Arrays.asList(header.split("\\s+,\\s+"));
     }
 
+    private String mockHttpResource(final String filename, final String type, final String content) {
+        wireMockServer.stubFor(head(urlEqualTo("/" + filename))
+                .willReturn(aResponse()
+                        .withHeader("Content-Length", Long.toString(content.length()))
+                        .withHeader("Content-Type", type)));
+        wireMockServer.stubFor(get(urlEqualTo("/" + filename))
+                .willReturn(aResponse()
+                        .withHeader("Content-Length", Long.toString(content.length()))
+                        .withHeader("Content-Type", type)
+                        .withBody(content)));
+
+        return "http://localhost:" + wireMockServer.port() + "/" + filename;
+    }
+
+    private String mockHttpResourceNoType(final String filename, final String content) {
+        wireMockServer.stubFor(head(urlEqualTo("/" + filename))
+                .willReturn(aResponse()
+                        .withHeader("Content-Length", Long.toString(content.length()))));
+        wireMockServer.stubFor(get(urlEqualTo("/" + filename))
+                .willReturn(aResponse()
+                        .withHeader("Content-Length", Long.toString(content.length()))
+                        .withBody(content)));
+
+        return "http://localhost:" + wireMockServer.port() + "/" + filename;
+    }
+
+    private String externalContentHeader(final String uri, final String handling) {
+        return externalContentHeader(uri, handling, null);
+    }
+
+    private String externalContentHeader(final String uri, final String handling, final String type) {
+        String link = "<" + uri + ">; rel=\"" + EXTERNAL_CONTENT_LINK_REL + "\"";
+        if (handling != null) {
+            link += "; handling=\"" + handling + "\"";
+        }
+        if (type != null) {
+            link += "; type=\"" + type + "\"";
+        }
+        return link;
+    }
+
+    private String createExternalBinary(final String uri, final String id, final String handling,
+            final String contentType) {
+        final Headers headers = new Headers(
+                new Header(LINK, externalContentHeader(externalUri, handling, contentType)),
+                new Header(SLUG, id));
```

Comment:
Please avoid using the `Slug` header where possible as support for it is not required AFAIK.